mayastor/src/lvm/lvm.rs

```diff
@@ -0,0 +1,285 @@
+use tokio::process::Command;
+use crate::lvm::error::Error;
+
+use rpc::mayastor::{CreatePoolRequest, Pool, PoolType, PoolState};
+
+const PVCREATE_COMMAND: &str = "pvcreate";
+const PVS_COMMAND: &str = "pvs";
+const PVREMOVE_COMMAND: &str = "pvremove";
+const VGCREATE_COMMAND: &str = "vgcreate";
+const VGS_COMMAND: &str = "vgs";
+const VGREMOVE_COMMAND: &str = "vgremove";
+
+pub async fn lookup_lvm_vg(name: &str) -> Option<Pool> {
+    let pools = list_lvm_vg().await.ok()?;
+    match pools.iter().find(|p| p.name == name) {
+        Some(p) => {
+            return Some(Pool{
+                name: p.name.as_str().to_owned(),
+                disks: p.disks.as_slice().to_owned(),
+                capacity: p.capacity,
+                used: p.used,
+                pooltype: PoolType::Lvm.into(),
+                state: PoolState::PoolOnline.into(),
+            });
+        }
+        None => None
+    }
+}
+
+pub async fn import_or_create_lvm_vg(req: CreatePoolRequest) -> Result<Pool, Error> {
+    let name = req.name.as_str();
+    let disks = req.disks.as_slice();
+    let pool = match import_lvm_vg(CreatePoolRequest{
```

Comment:
We could use 'Result::or_else()' combinator to handle the absence of the pool and eliminate intensive use of the 'match' operator.